client/forms/file-input.tsx

```diff
@@ -19,59 +19,90 @@ const ClearButton = styled(IconButton)`
   margin-left: 16px;
 `
 
-// Quick and ugly wrap for <input type='file'> that works with rest of the form stuff
-export function FileInput({
-  value,
-  allowErrors = false,
-  errorText,
-  inputProps,
-  onChange,
-  onFilesCleared,
-}: {
-  value?: File | File[] | null
+export interface FileInputHandle {
+  clear: () => void
+  click: () => void
+}
+
+interface FileInputProps {
+  value?: File | File[] | string | null
   allowErrors?: boolean
   errorText?: string
+  className?: string
   inputProps: React.InputHTMLAttributes<HTMLInputElement>
   onChange?: (file?: File | File[]) => void
   onFilesCleared?: () => void
-}) {
-  const { t } = useTranslation()
-  const inputRef = useRef<HTMLInputElement>(null)
+}
 
-  const onInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    const files = event.target.files ? Array.from(event.target.files) : []
-    onChange?.(inputProps.multiple ? files : files[0])
-  }
+// Quick and ugly wrap for <input type='file'> that works with rest of the form stuff
```

Comment:
Now that this has gone through like 4 separate modifications I am not sure this comment is really true